### ISSUE TITLE
Logical expressions parts should not span multiple lines

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#555]
We should issue a warning in case a specific logical expression spans multiple lines.

Following code should report a violation:
```c#
var flag = string.Compare("some string",
                           "some other string",
                           CultureInfo.InvariantCulture,
                           CompareOptions.IgnoreCase) ==
            0 || string.Compare("some string", "even another string", CultureInfo.InvariantCulture, CompareOptions.IgnoreCase) == 0;
```

The reason is because the first part spans multiple lines and is difficult to read (in comparison to the second part).

Following code should *NOT* report a violation:
```c#
var flag = string.Compare("some string", "some other string", CultureInfo.InvariantCulture, CompareOptions.IgnoreCase) == 0 ||
           string.Compare("some string", "even another string", CultureInfo.InvariantCulture, CompareOptions.IgnoreCase) == 0;
```
because now the first part is much easier to read and understand.

In addition, following code should also *NOT* report a violation:
```c#
if (a.B == null ||
    a.B.c == null ||
    a.B.c.D == null)
```
because each part is on a specific single line.
